test/screens/alerts/alert_test.exs

```diff
@@ -0,0 +1,88 @@
+defmodule Screens.Alerts.AlertTest do
+  use ExUnit.Case, async: true
+
+  alias Screens.Alerts.Alert
+
+  defp alert_json(id) do
+    %{
+      "id" => id,
+      "attributes" => %{
+        "active_period" => [],
+        "created_at" => nil,
+        "updated_at" => nil,
+        "cause" => nil,
+        "effect" => nil,
+        "header" => nil,
+        "informed_entity" => [],
+        "lifecycle" => nil,
+        "severity" => nil,
+        "timeframe" => nil
+      }
+    }
+  end
+
+  describe "fetch_by_stop_and_route/3" do
+    setup do
+      stop_based_alerts = [alert_json("1"), alert_json("2"), alert_json("3")]
+      route_based_alerts = [alert_json("4"), alert_json("3"), alert_json("5")]
+
+      stop_ids = ~w[1265 1266 10413 11413 17411]
+      route_ids = ~w[22 29 44]
+
+      stop_ids_param = Enum.join(stop_ids, ",")
+      route_ids_param = Enum.join(route_ids, ",")
+
+      %{
+        stop_ids: ~w[1265 1266 10413 11413 17411],
+        route_ids: ~w[22 29 44],
+        get_json_fn: fn
+          _, %{"filter[stop]" => ^stop_ids_param, "filter[route]" => ^route_ids_param} ->
+            {:ok, %{"data" => stop_based_alerts}}
+
+          _, %{"filter[route]" => ^route_ids_param} ->
+            {:ok, %{"data" => route_based_alerts}}
+        end,
+        x_get_json_fn1: fn
+          _, %{"filter[stop]" => ^stop_ids_param, "filter[route]" => ^route_ids_param} -> :error
+          _, %{"filter[route]" => ^route_ids_param} -> {:ok, %{"data" => route_based_alerts}}
+        end,
+        x_get_json_fn2: fn
```

Comment:
These tests might be easier to follow if the data/injections used by each one were defined within the test body, if it's only relevant to that test. With this "style" of setup, the data used by unrelated tests is mixed together, making it a bit more difficult to see what goes where. It's easy enough to follow in this test, where there are only two cases, but in `Widgets.AlertsTest` I think it's already revealing some [mystery guest](http://xunitpatterns.com/Obscure%20Test.html#Mystery%20Guest) issues.